File: networking/src/main/java/com/jbrunton/networking/repositories/HttpGenresRepository.java
package com.jbrunton.networking.repositories;

import com.jbrunton.entities.Genre;
import com.jbrunton.entities.GenresRepository;
import com.jbrunton.networking.resources.genres.GenresResponse;
import com.jbrunton.networking.services.LegacyMovieService;

import java.util.List;

import io.reactivex.Observable;

public class HttpGenresRepository implements GenresRepository {
    private final LegacyMovieService service;

    public HttpGenresRepository(LegacyMovieService service) {
        this.service = service;
    }

    @Override public Observable<List<Genre>> genres() {
        return service.genres()
                .map(GenresResponse::toCollection);
    }
}

File: networking/src/main/java/com/jbrunton/networking/repositories/HttpGenresRepository.kt
package com.jbrunton.networking.repositories

import com.jbrunton.entities.models.Genre
import com.jbrunton.entities.repositories.DataStream
import com.jbrunton.entities.repositories.GenresRepository
import com.jbrunton.entities.repositories.toDataStream
import com.jbrunton.networking.services.MovieService


class HttpGenresRepository(private val service: MovieService) : GenresRepository {
    override fun genres(): DataStream<List<Genre>> {
        return service.genres()
                .map { it.toCollection() }
                .toDataStream()
    }
}
